### ISSUE TITLE
Error when running from Jenkins

### ISSUE THREAD

**Issue by Eldadc** [Computician/PMCCommand#2]
Unhandled Exception : System.runtime.IteropServices.COMException: call was rejected bt calle 
at ENVDTE80.DTE2.set_UserControl (Boolean UserControl)
at PMCCommand.Program.GETDTE2() in
line 262
at PMCCommand.Programs.Execute()
Line 119

**Comment by benwtrent** [comment]
@Eldadc the call looks like it was rejected. Have you verified that the security settings are correct in `Control Panel - Administrative Tools - Component Services - Computers - My Computer - DCOM Config` ? You want change access for `Microsoft Visual Studio <version>` so that the remote executor has rights.

**Comment by Eldadc** [comment]
So what settings do I need to change.

**Comment by benwtrent** [comment]
@Eldadc its there in the comment. On the server change the DCOM config so that visual studio is allowed to have remote executors to call it. 

This stackoverflow answer talks about Microsoft Word, but the same concept applies (just for visual studio) https://stackoverflow.com/questions/1491123/system-unauthorizedaccessexception-retrieving-the-com-class-factory-for-word-in/2560877#2560877

Additionally, you may have to reboot the server for it to take effect.

**Comment by Eldadc** [comment]
Hi,

When you mentioned remote executors , Do you mean the user that run the
Jenkins slave ?

**Comment by benwtrent** [comment]
@Eldadc no. To run PMCCommand, two processes are communicating. By default, for certain users on the system, processes communicating with each other over COM is not allowed. You need to make sure that security settings allow one process to communicate with the visual studio process. This needs to be on the server that is going to run PMCCommand.
